src/LocateDependencies/LocateDependenciesViaComposer.php

```diff
@@ -38,12 +37,10 @@ public function __construct(
 
     public function __invoke(string $installationPath): SourceLocator
     {
-        Assert::file($installationPath . '/composer.json');
+        Psl\invariant(Filesystem\is_file($installationPath . '/composer.json'), 'Could not locate composer.json within installation path.');
 
         $this->runInDirectory(function () use ($installationPath): void {
-            $installer = ($this->makeComposerInstaller)($installationPath);
-
-            assert($installer instanceof Installer);
+            $installer = Type\object(Installer::class)->assert(($this->makeComposerInstaller)($installationPath));
```

Comment:
I think `$this->makeComposerInstaller` is missing an `@psalm-var callable(): Installer`? This assertion shouldn't be needed then...